src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -1658,25 +1658,20 @@ void C2_MacroAssembler::round_double_mode(FloatRegister dst, FloatRegister src,
 // otherwise return +/- 1.0 using sign of input.
 // one - gives us a floating-point 1.0 (got from matching rule)
 // bool is_double - specifies single or double precision operations will be used.
-void C2_MacroAssembler::signum_fp(FloatRegister dst, FloatRegister src, FloatRegister one, bool is_double) {
-  Register tmp1 = t0;
-
+void C2_MacroAssembler::signum_fp(FloatRegister dst, FloatRegister one, Register tmp, bool is_double) {
```

Comment:
I think we could still make use of scratch register `t0` in this function. That will save us one GP register.

Reply:
Sure, it make sense.